### PR TITLE
Flags w env var fallbacks

### LINKED ISSUE #4: Accept site as an env var
In order to configure which site to check and to be a good "cloud-native citizen", the CLI should read sites in via environment variables. Presumably, checking more than one site will be desirable (or do we want to have one blackbox instance dedicated to one site? 🤔), so a list of URLs would be useful.

My concern here is that the site domain entity could become more complex as this gets built out and a single env var for the URL, while covering this issue now, may not be what we need down the road. Immediately, I can imagine wanting to set frequency per site, having various URIs per site (these could just be different sites though), setting cookies on the request for allow list purposes on protected endpoints? Maybe the response to this should be: YAGNI? 🤔

### PR DESCRIPTION
**Pull request by BrianGreenhill**
fixes #4

## TODO

- [x] have environment variables stand in as defaults for the flags
- [x] fall back to default schedule if env var missing

### CODE REVIEW

**Review by wormi4ok — COMMENTED** [review]
I've heard that standard `flag` library is very limited, but this result looks better than I expected:

```
$ ./blackbox --help
Usage of ./blackbox:
  -name string
    	The name of the check (default "check_2021-02-07_14:32")
  -schedule string
    	The schedule in six star format (default "* * * * * *")
  -url string
    	The URL to check
```

File: main.go [hunk 1]
```diff
@@ -14,32 +15,44 @@ import (
...
-const URL = "https://www.google.com"
-const ScheduleFormat = "* * * * * *"
+var (
+	url      string
+	schedule string
+	name     string
+)
+
+func init() {
+	flag.StringVar(&name, "name", os.Getenv("TARGET_NAME"), "The name of the check")
```
Comment (wormi4ok) [inline main.go:25]: If I run the application without `TARGET_NAME` set in the environment, the name will be an empty string.

**Review by wormi4ok — COMMENTED** [review]
I'm not sure how to implement this nicely, but as a user, I expect the following precedence:
* CLI Argument 
* Environment variable
* Default value

Do you have the same expectation or is it only me?

File: main.go (hunk 1 above)
Comment (BrianGreenhill) [reply main.go:25]: yes, i saw this as well. we don't use the name yet so i didn't require it.

**Comment by BrianGreenhill** [comment]
> I'm not sure how to implement this nicely, but as a user, I expect the following precedence:
> 
> * CLI Argument
> * Environment variable
> * Default value
> 
> Do you have the same expectation or is it only me?

i'm not sure what a default value for a url could be, but for the schedule we could fallback to `* * * * * *` if both cli and env var aren't set. i think then we've fulfilled the spec you mention here. what do you think? is that a possible solution?

**Comment by BrianGreenhill** [comment]
🏓  @wormi4ok  :p

**Review by wormi4ok — APPROVED** [review]
PM approved 😁

**Pull request merged**